homeassistant/components/tplink/__init__.py

```diff
@@ -228,6 +228,13 @@ def legacy_device_id(device: Device) -> str:
     return device_id.split("_")[1]
 
 
+def get_device_name(device: Device) -> str:
+    """Get a name for the device. alias can be none on some devices."""
+    if device.alias:
+        return device.alias
+    return f"Unnamed {device.model}"
```

Comment:
Would this approach also fix #114752? The issue that aimed to solve was "partially" provisioned power strips where the individual sockets were not named, leaving the children with an empty name.

Reply:
Yes it would fix that